build/madcamp.xml

```diff
@@ -0,0 +1,94 @@
+      <property name="repo.dir" value="${pantheon.dest}" />
+      <property name="repo.source" value="${pantheon.repo}" />
+    </phing>
+
+    <phing target="init-branch" phingfile="vendor/palantirnet/the-build/tasks/lib/repositories.xml" inheritAll="false" dir=".">
+      <property name="repo.dir" value="${pantheon.dest}" />
+      <property name="repo.branch" value="${pantheon.branch}" />
+    </phing>
+  </target>
+
+
+  <!-- @see the-build/tasks/lib/artifacts.xml:run-composer-elsewhere -->
+  <target name="madcamp-copy-files-to-artifact">
+    <fail unless="pantheon.dest" />
+    <fail unless="pantheon.source" />
+
+    <!-- Remove all the existing files so that we can install cleanly. -->
+    <delete includeemptydirs="true">
+      <fileset dir="${pantheon.dest}" excludes=".git,.git/**" />
+    </delete>
+
+    <!--
+        also, we need to manually copy the custom code into the artifact.
+        doing it based on what is checked in to git prevents us from
+        accidentally adding local files or settings.php in to the
+        build repository.
+
+        AND, drupal has applied permissions to the sites directory that
+        prevent us from writing to it, so we need to squash them.
+    <exec command="chmod 777 www/sites/default" dir="${build.dir}" />
+        -->
+
+    <tempfile property="tmpfile" destdir="${build.dir}/artifacts" />
+    <exec command="git ls-files" dir="${build.dir}/${pantheon.source}" output="${tmpfile}" />
+
+    <copy todir="${pantheon.dest}" overwrite="true" haltonerror="true">
+      <filelist dir="${build.dir}/${pantheon.source}" listfile="${tmpfile}" />
+    </copy>
+
+    <delete file="${tmpfile}" />
+
+    <!--
+        having git repositories in the vendor dir messes up the build
+        artifact, since the entire vendor dir gets checked in to the
+        artifact and git will automatically try to do submodules for git
+        directories like this. so we strip them.
+        -->
+    <delete includeemptydirs="true">
+      <fileset dir="${pantheon.dest}" defaultexcludes="false">
+        <include name="*/**/.git/**" />
+        <include name="*/**/.git" />
+      </fileset>
+    </delete>
+  </target>
+
+
+  <target name="madcamp-commit-artifact-changes">
+    <fail unless="pantheon.dest" />
+    <fail unless="pantheon.tag_prefix" />
+    <fail unless="pantheon.branch" />
+
+    <phing target="commit" phingfile="vendor/palantirnet/the-build/tasks/lib/repositories.xml" inheritAll="false" dir=".">
+      <property name="build.dir" value="${build.dir}" />
+      <property name="repo.dir" value="${pantheon.dest}" />
+      <property name="tag_prefix" value="${pantheon.tag_prefix}" />
+      <property name="message" value="Drupal artifact." />
+    </phing>
+
+    <phing target="push" phingfile="vendor/palantirnet/the-build/tasks/lib/repositories.xml" inheritAll="false" dir=".">
```

Comment:
This will always push, regardless of the value of the `push` property... so if you run `vendor/bin/phing deploy -Dpush=n`, this script will still try to push. You'll need to wrap this in an `<if>` statement in order to respect that flag.